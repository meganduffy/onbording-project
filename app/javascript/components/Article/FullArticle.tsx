import React from 'react';

export interface FullArticleProps {
    title:string,
    content:string,
    first_name:string,
    last_name:string,
    created_at:string
}

const FullArticle: React.FC<FullArticleProps> = (props) => {
    return (
        <div className="articles-details">
            <h1 className="articles-title">{props.title}</h1>

            <p>{props.content}</p>

            <div className="articles-details-info">
                <p>{props.first_name} {props.last_name}</p>
                <p>{props.created_at}</p>
            </div>

            
        </div>
    )
}

export default FullArticle;