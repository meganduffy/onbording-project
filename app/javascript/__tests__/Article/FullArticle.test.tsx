import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import FullArticle from '../../components/Article/FullArticle';

let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', ()=>{
  const tree = renderer.create(<FullArticle title='New Post' content='Ipsum Lorem'
  first_name='Megan' last_name='Duffy' created_at='27 March 2020' />).toJSON();
  expect(tree).toMatchSnapshot();
});

it( 'renders with a title', ()=>{
    act( ()=>{render(<FullArticle title='New Post' />, container)} )
        expect(container.textContent).toBe("New Post ")
});
