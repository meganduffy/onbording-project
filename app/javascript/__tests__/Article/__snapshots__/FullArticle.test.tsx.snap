// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="articles-details"
>
  <h1
    className="articles-title"
  >
    New Post
  </h1>
  <p>
    Ipsum Lorem
  </p>
  <div
    className="articles-details-info"
  >
    <p>
      Megan
       
      Duffy
    </p>
    <p>
      27 March 2020
    </p>
  </div>
</div>
`;
